### DISCUSSION TITLE
[RFC] In browser chat bot.

### DISCUSSION THREAD

**Discussion by bdougie** [open-sauced/open-sauced#682] [Ideas]
@yg create a [test PR leveraging Crisp chat](https://github.com/open-sauced/open-sauced/pull/636). 

![crsip-chat](https://user-images.githubusercontent.com/61746440/83963847-70579380-a8c6-11ea-98bf-d55072986f07.png)

# Why?

## Support
A chat integration could provide the ability for this project to get direct feedback from its users. My thought is to provide an @open-sauced/support team access to interact with users. This chat would also link to a private discord channel where the team could interact with the conversation.

Some support requests examples:
1. How do I get started?
2. What are some good projects to contribute to?
3. I found a bug (it is not obvious that someone could open an issue in the browser today). 

## Scheduled notifications
A chat integration could provide notifications about new releases and breaking changes. 

## Why not just write more code?
It is quite possible that this could all be done in code and there is no need for integrating a 3rd party tool. 

Is there a to add a webhook to the discord channel in discord and have a response a way to automate the way a support team would have access to this service.

# Potential implementation

1. Created a new discussion in GitHub
2. The bot should create a new private channel in Discord under support category
3. The bot platform should be synchronous on both GitHub and Discord
4. If the discussion is marked as answer, the channel in discord should get deleted

Leveraging webhooks and [repository_dispatch](https://developer.github.com/webhooks/event-payloads/#repository_dispatch) to update the issue comment thread from discord/crisp. 

## Downsides to Crisp
- Customizable with a paid team
- Free for only 2 users.

### We are open to other suggestions and experiences. Please reply below with thoughts.

**Comment by NotMoni** [comment]
Thumbs up on this post for the Discord Bot. 

One downside.

its needs to be hosted 24/7 
even a Raspberry Pi 2 would suffice with node 13 above.

**Reply by bdougie** [reply]
I have a ton of azure credits I could use if hosting is needed. I not familiar with discord bots, but assume they integrate similar to any other bots, i.e. [hubot](https://hubot.github.com/).

**Reply by NotMoni** [reply]
well that is good, so if you want this do a 👍

**Comment by bdougie** [comment]
Looking at https://feedback.fish/ now. Max and the team built it in 3 days on productweekend.live.

**Comment by 0-vortex** [comment]
There used to be some popular services floating around like https://widgetbot.io and https://elfsight.com promising forum like witch-boxes. These operate on top of the generic discord widget by proxying a bot user and rewriting messages to enable discord users outside the platform to send and view messages from the website. Having some experience with bot framework doing that via OAuth, it becomes a chore to filter out spam but we could also look at discord server embed FOSS like [TitanEmbeds/Titan](https://github.com/TitanEmbeds/Titan) - I know this discussion was originally for website/support interaction but we could create a popout style embed and potentially ditch the twitch chat with the streaming channels on discord, something superior in many ways IMO

**Comment by 0-vortex** [comment]
set up a titanembed on a custom server and it somewhat gets the job done. advantage that the software is available on github means we could potentially host the server and have close to 0 delay in communication. styling the embed to a support widget would not really fit our purpose, as the use case described here is more closely related to creating a new `support-{guestTag}-timestamp` channel, where the bot would ping the triage team. The implementation is simple but requires we go back to the drawing board and create a proper `open-sauced/discord-bot`

going back to titanembed if you want to give it a try, default ui, configured with just a custom channel looks somewhat like this:

<img width="1414" alt="Screenshot 2021-08-08 at 21 26 04" src="https://user-images.githubusercontent.com/237133/128643573-3a0026b5-0728-478c-b6ec-d8144fe7e080.png">
<img width="1414" alt="Screenshot 2021-08-08 at 21 27 19" src="https://user-images.githubusercontent.com/237133/128643578-0b720ddb-ed49-4bdf-8406-b9759b32bcb1.png">
<img width="1414" alt="Screenshot 2021-08-08 at 21 27 34" src="https://user-images.githubusercontent.com/237133/128643580-0b98b5d3-2a9a-4245-9622-aba4f06cd3d3.png">
